src/providers/token-fee-fetcher.ts

```diff
@@ -0,0 +1,117 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { BaseProvider } from '@ethersproject/providers';
+import { ChainId } from '@uniswap/sdk-core';
+
+import { TokenFeeDetector__factory } from '../types/other/factories/TokenFeeDetector__factory';
+import { TokenFeeDetector } from '../types/other/TokenFeeDetector';
+import { log, WRAPPED_NATIVE_CURRENCY } from '../util';
+
+import { ProviderConfig } from './provider';
+
+const DEFAULT_TOKEN_BUY_FEE_BPS = BigNumber.from(0);
+const DEFAULT_TOKEN_SELL_FEE_BPS = BigNumber.from(0);
+
+// on detector failure, assume no fee
+export const DEFAULT_TOKEN_FEE_RESULT = {
+  buyFeeBps: DEFAULT_TOKEN_BUY_FEE_BPS,
+  sellFeeBps: DEFAULT_TOKEN_SELL_FEE_BPS,
+};
+
+type Address = string;
+
+export type TokenFeeResult = {
+  buyFeeBps?: BigNumber;
+  sellFeeBps?: BigNumber;
+};
+export type TokenFeeMap = Record<Address, TokenFeeResult>;
+
+// address at which the FeeDetector lens is deployed
+const FEE_DETECTOR_ADDRESS = (chainId: ChainId) => {
+  switch (chainId) {
+    case ChainId.MAINNET:
+    default:
+      return '0x57eC54d113719dDE9A90E6bE807524a86560E89D';
```

Comment:
Just noticed that interface uses 0x19C97dc2a25845C7f9d1d519c8C2d4809c58b43f. Need to follow-up with protocol folks on which contract is the right one.

Reply:
Most likely 0x19C97dc2a25845C7f9d1d519c8C2d4809c58b43f. 0x19C97dc2a25845C7f9d1d519c8C2d4809c58b43f don't have the `UniswapV2: Locked` error.